File: Server/src/main/content/global/plugins/item/BirdNestPlugin.java
package content.global.plugins.item;

import content.data.tables.BirdNestDropTable;
import core.game.interaction.OptionHandler;
import core.game.node.Node;
import core.game.node.entity.player.Player;
import core.game.node.item.Item;
import core.plugin.Plugin;

/**
 * Handles the searching of a bird nest item.
 */
public final class BirdNestPlugin extends OptionHandler {

	@Override
	public Plugin<Object> newInstance(Object arg) throws Throwable {
		for (BirdNestDropTable nest : BirdNestDropTable.values()) {
			nest.getNest().getDefinition().getHandlers().put("option:search", this);
		}
		return null;
	}

	@Override
	public boolean handle(Player player, Node node, String option) {
		final Item item = (Item) node;
		BirdNestDropTable.forNest(item).search(player, item);
		return true;
	}

	@Override
	public boolean isWalk() {
		return false;
	}
}

File: Server/src/main/content/global/handlers/item/plugin/BirdNestPlugin.kt
package content.global.handlers.item.plugin

import content.data.tables.BirdNestDropTable
import core.game.interaction.OptionHandler
import core.game.node.Node
import core.game.node.entity.player.Player
import core.game.node.item.Item
import core.plugin.Plugin

/**
 * Bird nest plugin
 *
 * @constructor Bird nest plugin
 */
class BirdNestPlugin : OptionHandler() {

    override fun newInstance(arg: Any?): Plugin<Any?>? {
        BirdNestDropTable.values().forEach { nest ->
            nest.nest.definition.handlers["option:search"] = this
        }
        return null
    }

    override fun handle(player: Player, node: Node, option: String): Boolean {
        val item = node as Item
        BirdNestDropTable.forNest(item)!!.search(player, item)
        return true
    }

    override fun isWalk(): Boolean {
        return false
    }
}
